### ISSUE TITLE
Unsupported Int 2F/AX=4309h?

### ISSUE THREAD

**Issue by JayQ2K** [joncampbell123/dosbox-x#2297]
Working through some EMM setups and XMSSTAT from the JemmEx 5.79B package I found that Int 2F/AX=4309h is not supported when starting DOSBox-X with the following settings:
xms                     = true
ems                     = false
umb                     = true
device = EMM386.EXE NOEMS

If you after launch go to XMSSTAT it shows this:
Int 2Fh, ax=4309h failed!

Using either HIMEM.SYS or HimemX.EXE it gives information about the XMS Handle Table.
Maybe it's not a really important thing, but upon seeing it I figured to share it.

**Comment by JayQ2K** [comment]
MSD.EXE (Microsoft Diagnostics) shows that the XMS Driver Version for xms=true is Version 3.01, while Int 2F/AX=4309h is something for HIMEM3.09+ so that might be the culprit alltogether....?

**Comment by joncampbell123** [comment]
INT 2F AX=4309h is a call to get the XMS handle table.

http://www.ctyme.com/intr/rb-4767.htm

**Comment by JayQ2K** [comment]
Indeed. Aware of that. It's moreover that I saw that the handle doesn't
give any results apart from it not being supported.
Again, might not be the most important thing, but since the diagnostics
tools gave this result I figured to share it.

**Comment by JayQ2K** [comment]
To build on this; it seems that any EMM that supports dynamic memory allocation (e.g. JEMM386) requires a handle array. Looking at INT 2F AX=4309h it mentions "pointer to XMS handle array" and thus is probably what you need to get dynamic memory allocation from your EMM.

For now this effectively means it is required to switch off dynamic memory allocation for any EMM with this support.

**Comment by JayQ2K** [comment]
Using the internal XMM leads to issues when launching an EMM.
The complete range I=C500-F7FF must be included as a switch for any EMM to ensure this address space becomes usable.
Without this statement the page frame space (E000-EFFF) is blocked from usage, meaning EMS activation is impossible.

Also when using internal XMM and <PATH>\EMM386.EXE I=C500-F7FF NOEMS VERBOSE the machine crashes into debug. This is related to the NOEMS statement since when I remove that the machine boots succesfully.

Using an external XMM (e.g. HIMEM) the above 2 issues don't arise.
I start to think this whole Int 2F/AX=4309h is quite important.
